app/src/main/java/cm/aptoide/pt/view/MainActivity.java

```diff
@@ -119,6 +142,49 @@ private void setupUpdatesNotification() {
     updatesBadge.setVisibility(View.GONE);
   }
 
+  @Override public Observable<PermissionService> autoUpdateDialogCreated() {
+    return autoUpdateDialogSubject;
+  }
+
+  @Override public void requestAutoUpdate() {
+    AlertDialog.Builder dialogBuilder = new AlertDialog.Builder(this);
+    final AlertDialog updateSelfDialog = dialogBuilder.create();
+    updateSelfDialog.setTitle(getText(R.string.update_self_title));
+    updateSelfDialog.setIcon(R.mipmap.ic_launcher);
+    updateSelfDialog.setMessage(
+        AptoideUtils.StringU.getFormattedString(R.string.update_self_msg, resources, marketName));
+    updateSelfDialog.setCancelable(false);
+    updateSelfDialog.setButton(DialogInterface.BUTTON_POSITIVE, getString(android.R.string.yes),
+        (arg0, arg1) -> {
+          autoUpdateDialog = new ProgressDialog(this);
+          autoUpdateDialog.setMessage(getString(R.string.retrieving_update));
+          autoUpdateDialog.show();
+          autoUpdateDialogSubject.onNext(this);
+        });
+    updateSelfDialog.setButton(Dialog.BUTTON_NEGATIVE, getString(android.R.string.no),
+        (dialog, arg1) -> {
+          dialog.dismiss();
+        });
+    if (is_resumed()) {
+      updateSelfDialog.show();
+    }
+  }
+
+  @Override public void handleAutoUpdateResult(boolean installFailed) {
+    if (installFailed) {
+      Snackbar.make(findViewById(android.R.id.content), R.string.unknown_error,
+          Snackbar.LENGTH_SHORT)
+          .show();
+    }
+    dismissDialog();
```

Comment:
We should have a method to show unknow error and a method to hide the dialog. Presenter would choose what to call based on the install result and we could remove this boolean from the method signature.